File: 0541-reverse-string-ii/0541-reverse-string-ii.cpp
class Solution {
public:
    string reverseStr(string s, int k) {
        int n = s.size();
        for(int x = 0; x < n; x+=2*k){
            int i = x;
            int j = min(x + k - 1, n - 1);
            while(i < j){
                char c = s[i];
                s[i] = s[j];
                s[j] = c;
                i++;
                j--;
            }
        }
        return s;
    }
};

File: 0541-reverse-string-ii/0541-reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {
        char[] c = s.toCharArray();
        for(int i = 0; i < c.length; i += 2 * k){
            int l = i;
            int r = Math.min(i + k - 1, c.length - 1);

            while(l < r){
                char a = c[l];
                c[l] = c[r];
                c[r] = a;
                l++;
                r--;
            }
        }
        return new String(c);
    }
}